src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -567,10 +579,15 @@ void TemporalMemory::activateDendrites(bool learn,
     NTA_CHECK( extraWinners.size() != 1 || extraWinners[0] != SENTINEL )
         << "TM.ActivateDendrites() missing argument extraWinners!";
 
+    //add extra active
+    auto& activeVec = activeCells_.getSparse();
     for(const auto &active : extraActive) {
       NTA_ASSERT( active < extra_ );
-      activeCells_.push_back( static_cast<CellIdx>(active + numberOfCells()) );
+      activeVec.push_back( static_cast<CellIdx>(active + numberOfCells()) ); 
```

Comment:
@ctrl-z-9000-times I'm having trouble with the extraActive hack here, I recall we discussed the API and this seemed like a good compromise.  
- if I use `SDR activeCells_`, it checks for validity of the new sparse indices, thus triggers here. 
- You've mentioned you'll be pulling back the TM's extraActive/Winners back to CP/Connections? 
- the `active + numberOfCells()` does not have meaning in SDR's context, where is this interpreted? 

How to proceed? 
- I could revert the changes to TM.activeCells_ back to vector. 
- the `extra*` make the API still inconvenient, should we use `TM.setExtraActive/Winners(SDR)` instead?

Reply:
removing the change from this PR to make it pass, will reintroduce in a consequent PR.

Reply:
> the extra* make the API still inconvenient

I should have named it a lot more verbosely.  Maybe "externalPredictiveInputs"?

> if I use SDR activeCells_, it checks for validity of the new sparse indices, thus triggers here.

The external predictive inputs are not part of the active cells of this TM, and shouldn't be added to the activeCells.  This is a hack, really it should concatenate both the activeCells and the externalPredictiveCells into a new vector, instead of repurposing an existing vector.

> the active + numberOfCells() does not have meaning in SDR's context, where is this interpreted?

That code concatenates the `activeCells` with the `extra` cells.  By offsetting all of the `extra` cell activations to past the end of the `activeCells` SDR, this keeps both sources of presynaptic input separate/non-overlapping from each other.

> You've mentioned you'll be pulling back the TM's extraActive/Winners back to CP/Connections?

No.  What I'd like to do is change the Connections class to accept an SDR of presynaptic inputs.  Currently it accepts a list of sparse indexes.

Reply:
I'm opening a PR with the getActiveCells() changes, please help me resolve the issue there.